File: Data Structures & Algorithms/generate-parentheses/submission-2.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        ans = []

        def generate(left: int = 0, right: int = 0, sub: str = "") -> None:
            if len(sub) == n * 2:
                ans.append(sub)
                return
            if left < n:
                generate(left + 1, right, sub + "(")
            if right < left:
                generate(left, right + 1, sub + ")")

        generate()

        return ans

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.js
class Solution {
	/**
	 * @param {TreeNode} root
	 * @return {number[][]}
	 */
	levelOrder(root) {
		if (!root) return [];
		const q = [root];
		const ans = [];
		let fixed = 1;
		while (q.length) {
			if (q.length % fixed === 0) {
				ans.push(Array.from(q, (el) => el.val));
				fixed += fixed;
			}
			const top = q.shift();
			if (top.left)
				q.push(top.left);

			if (top.right)
				q.push(top.right);
		}
		return ans;
	}
}
